### PR TITLE
[PROCS-3838] Add handling for ECS Fargate misconfiguration during chunking

### PR DESCRIPTION
**Pull request by daniel-taf**
### What does this PR do?

This PR handles the issue described in https://github.com/DataDog/datadog-agent/issues/22940. Previously, when on ECS Fargate with a misconfigured pidMode setting, the process agent can fails to maintain the relationship between `chunks` and `props` in the ChunkAllocator when calling `appendContainerWithoutProcesses` causing a panic when `props` is accessed during `ChunkAllocator.Accept`. Now, `ChunkAllocator.Accept` is used directly in the function to that `props` and `chunks` remain equal in length.

### Motivation

PROCS-3838

### Additional Notes

The added unit test confirmed the behavior as it results in a panic on `main` but passes with this PR's changes.

### Describe how to test/QA your changes

Run the agent as a sidecar container with container on ECS Fargate without pidMode set. Verify the panic no longer occurs and the log warning appears. Also check that process/container data is available on Datadog.

Sample task definition to create service from:
```
{
    "containerDefinitions": [
            {
                "name": "redis",
                "image": "redis:latest",
                "cpu": 0,
                "portMappings": [],
                "essential": true,
                "environment": [],
                "mountPoints": [],
                "volumesFrom": [],
                "dockerLabels": {
                    "com.datadoghq.ad.check_names": "[\"redisdb\"]",
                    "com.datadoghq.ad.init_configs": "[{}]",
                    "com.datadoghq.ad.instances": "[{\"host\": \"%%host%%\", \"port\": 6379}]"
                },
                "systemControls": []
            },
            {
                "name": "datadog-agent",
                "image": "public.ecr.aws/datadog/agent:latest",
                "cpu": 0,
                "portMappings": [],
                "essential": true,
                "environment": [
                    {
                        "name": "ECS_FARGATE",
                        "value": "true"
                    },
                    {
                        "name": "DD_PROCESS_AGENT_PROCESS_COLLECTION_ENABLED",
                        "value": "true"
                    }
                ],
                "mountPoints": [],
                "volumesFrom": [],
                "secrets": [
                    {
                        "name": "DD_API_KEY",
                        "valueFrom": "<secret>"
                    }
                ],
               ...
            }
        ],
}
```
Log:
```
2024-04-25 20:06:47 UTC | PROCESS | WARN | (pkg/process/checks/chunking.go:25 in chunkProcessesBySizeAndWeight) | No processes found for container , pidMode may not be configured correctly (set to task)	
```

### CODE REVIEW

File: pkg/process/checks/chunking.go
Comment (robertjli) [inline pkg/process/checks/chunking.go:23]:
```suggestion
		// c) if no processes were collected from the container (e.g. pidMode not set to "task" on ECS Fargate)
```

Optional, I'd recommend generalizing the scenario in case it comes up in other situations as well

**Review by robertjli — APPROVED** [review]

File: pkg/process/checks/chunking.go
Comment (daniel-taf) [reply pkg/process/checks/chunking.go]: Done

File: releasenotes/notes/process-fargate-chunking-pidMode-0144d9e0833cb19b.yaml
Comment (cswatt) [inline releasenotes/notes/process-fargate-chunking-pidMode-0144d9e0833cb19b.yaml:11]:
```suggestion
    The Process Agent no longer crashes when pidMode is misconfigured on ECS Fargate. A warning is logged instead.
```

**Review by cswatt — APPROVED** [review]
just a grammar suggestion on the changelog

approving so I don't block merge

**Pull request merged**
